hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -642,9 +645,7 @@ int SaraNcpClient::queryAndParseAtCops(CellularSignalQuality* qual) {
     cgi_.mobile_network_code = static_cast<uint16_t>(::atoi(mobileNetworkCode));
 
     if (ncpId() == PLATFORM_NCP_SARA_R410 || ncpId() == PLATFORM_NCP_SARA_R510) {
-        if (act == particle::to_underlying(CellularAccessTechnology::LTE)) {
-            act = particle::to_underlying(CellularAccessTechnology::LTE_CAT_M1);
-        }
+        act = particle::to_underlying(CellularAccessTechnology::LTE_CAT_M1);
```

Comment:
Seems like the easiest way to avoid this problem is to ignore the AcT parameter for R410/510 and always fix it to LTE_CAT_M1

Reply:
This somewhat hard codes the RAT for R410 as well which doesn't have the -1 bug.  The bug also will be resolved in a future update from u-blox for the R510.  If we hard code it this way, would it show that we are always connected to Cat-M1, even when we might not have signal?  For the previous R410 code, it checked if the AcT was LTE first, before setting to Cat-M1.

I believe the fix in the parser should be good enough to resolve this for R510
```
if (modemIsSaraRxFamily() && d == -1 ) {
    d = UBLOX_SARA_RAT_LTE_CAT_M1;
}
```

Reply:
Follow up... just caught that this was Gen 3 vs. Gen 2 code I referenced.  Now I'm also wondering if we have this R410 "fix" in Gen 2.

So I'm thinking we should not change this code here for Gen 3, but instead test for -1 and hard code to Cat-M1 for R510 only.

Reply:
Changed to explicitly check for act == -1 and R510 modem